Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,369 @@
+                switch (tag.Key)
+                {
+                    case "operation.name": // not defined by OpenTracing
+                        {
+                            telemetry.Name = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "operation.type": // not defined by OpenTracing
+                        {
+                            telemetry.Type = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "operation.data": // not defined by OpenTracing
+                        {
+                            telemetry.Data = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "component":
+                        {
+                            component = tag.Value;
+                            break;
+                        }
+                    case "db.statement":
+                        {
+                            dbStatement = tag.Value;
+                            break;
+                        }
+                    case "error":
+                        {
+                            bool failed;
+                            if (bool.TryParse(tag.Value, out failed))
+                            {
+                                telemetry.Success = !failed;
+                                continue; // skip Properties
+                            }
+                            break;
+                        }
+                    case "http.status_code":
+                        {
+                            telemetry.ResultCode = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "http.url":
+                        {
+                            httpUrl = tag.Value;
+                            if (Uri.TryCreate(tag.Value, UriKind.RelativeOrAbsolute, out requestUri))
+                            {
+                                continue; // skip Properties
+                            }
+                            break;
+                        }
+                    case "peer.address":
+                        {
+                            peerAddress = tag.Value;
+                            break;
+                        }
+                    case "peer.hostname":
+                        {
+                            telemetry.Target = tag.Value;
+                            continue; // skip Properties
+                        }
+                    case "peer.service":
+                        {
+                            peerService = tag.Value;
+                            break;
+                        }
+                }
+
+                // if more than one tag with the same name is specified, the first one wins
+                if (!telemetry.Properties.ContainsKey(tag.Key))
```

Comment:
I think this part should live in base SDK OperationCorrelationTelemtryInitializer. This way it would work for Activities created without DiagnosticSource events or non-generic listeners.